core/src/main/scala/com/databricks/labs/remorph/parsers/intermediate/dml.scala

```diff
@@ -38,13 +38,52 @@ case class UpdateTable(
   override def output: Seq[Attribute] = target.output
 }
 
-case class MergeTables(
+/**
+ * The logical plan of the MERGE INTO command, aligned with SparkSQL
+ */
+case class MergeIntoTable(
     target: LogicalPlan,
-    source: Option[LogicalPlan],
-    conditions: Option[Expression],
+    source: LogicalPlan,
+    mergeCondition: Expression,
+    matchedActions: Seq[MergeAction],
+    notMatchedActions: Seq[MergeAction],
+    notMatchedBySourceActions: Seq[MergeAction],
     outputRelation: Option[LogicalPlan],
     options: Option[Expression])
```

Comment:
has to be

```
case class MergeIntoTable(
    targetTable: LogicalPlan,
    sourceTable: LogicalPlan,
    mergeCondition: Expression,
    matchedActions: Seq[MergeAction] = Seq.empty,
    notMatchedActions: Seq[MergeAction] = Seq.empty)
```

everything else should go to extensions.

Reply:
NotMatchedBySourceActions is standard and in Spark/Catalyst - outputRelation is TSQL Specific however, so I will move it. Catalyst definition is:

```scala
case class MergeIntoTable(
    targetTable: LogicalPlan,
    sourceTable: LogicalPlan,
    mergeCondition: Expression,
    matchedActions: Seq[MergeAction],
    notMatchedActions: Seq[MergeAction],
    notMatchedBySourceActions: Seq[MergeAction],
    withSchemaEvolution: Boolean) extends ...
 ```
 
As far as I know only SparkSQL supports `WITH SCHEMA EVOLUTION` on MERGE, so I think we can leave that out.

Reply:
Done.